programs/bpf_loader/src/lib.rs

```diff
@@ -298,6 +298,20 @@ fn check_loader_id(id: &Pubkey) -> bool {
         || bpf_loader_upgradeable::check_id(id)
 }
 
+fn calculate_heap_cost(heap_size: u64, heap_cost: u64, enable_rounding_fix: bool) -> u64 {
+    const KILOBYTE: u64 = 1024;
+    const PAGE_SIZE_KB: u64 = 32;
+    let mut rounded_heap_size = heap_size;
+    if enable_rounding_fix {
+        rounded_heap_size = rounded_heap_size
+            .saturating_add(PAGE_SIZE_KB.saturating_sub(1).saturating_mul(KILOBYTE));
```

Comment:
Shouldn't it be `PAGE_SIZE_KB.saturating_mul(KILOBYTE).saturating_sub(1)` (order of mul and sub swapped)?

Reply:
The math should be `(heap_size + (32KB-1))/32KB)`, thanks for catching it.